add begin/end to ExecuTorch pytree::arr

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #7655
* #7654
* __->__ #7653

Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

Commit:
add begin/end to ExecuTorch pytree::arr

Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

[ghstack-poisoned]

swolchok committed:
Update on "add begin/end to ExecuTorch pytree::arr"


Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

[ghstack-poisoned]

dbort reviewed (approved):
Seems ok, but please add some unit tests for this new functionality

swolchok committed:
Update on "add begin/end to ExecuTorch pytree::arr"


Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

[ghstack-poisoned]

swolchok committed:
Update on "add begin/end to ExecuTorch pytree::arr"


Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

[ghstack-poisoned]

swolchok committed:
Update on "add begin/end to ExecuTorch pytree::arr"


Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

[ghstack-poisoned]

Commit:
Update on "add begin/end to ExecuTorch pytree::arr"


Allows use of range-for.

Differential Revision: [D68166302](https://our.internmc.facebook.com/intern/diff/D68166302/)

[ghstack-poisoned]
